### ISSUE TITLE
update CucumberJSAllureReporter to support latest cucumber "@cucumber/cucumber"

### ISSUE THREAD

**Issue by Alje001** [allure-framework/allure-js#246]
HI;

we updated our testproject to support @cucumber/cucumber as the old version is deprecated. See following Links:
deprecated: https://www.npmjs.com/package/cucumber
current cucumber : https://www.npmjs.com/package/@cucumber/cucumber 

And with the latest version we are facing the issue that CucumberJSAllureReporter is importing cucumber dependecies from the deprecated cucumber. 

`> import { Formatter, World as CucumberWorld } from "cucumber";

That is why I am not getting allure reports anymore. 

Would you please update CucumberJSAllureReporter to support the latest @cucumber/cucumber

thank you for supporting

**Comment by Ivan-Katovich** [comment]
Have the same issue, are you going to resolve the issue in the nearest future? 
The issue blocks starting allure as main reporting systems for any new projects (my case)

**Comment by fescobar** [comment]
Also, it's necessary to have in mind that the new allure reporter supports executions in parallel. Currently is not supporting that.
https://github.com/korobochka/cucumberjs-allure2-reporter/issues/6

**Comment by alfflores** [comment]
same topic https://github.com/allure-framework/allure-js/issues/227

**Comment by baev** [comment]
btw, I already started my work on that https://github.com/allure-framework/allure-js/pull/346 . Feel free to contribute to cucumberjs7 branch as well, that will speedup a progress a bit.

From my side I expect to finish the work by the end of September (I'm pretty busy on my schedule :( )

**Comment by baev** [comment]
I'll close this one as duplicate of #227
